tests/rest/devices.spec.js

```diff
@@ -144,21 +265,29 @@ describe('Devices interface', () => {
 
   describe('GET /devices:uuid', function() {
     it('should return a single object', (done) => {
-      chai.request(server)
-        .get('/devices/' + test_uuid)
-        .end((err, res) => {
-          should.not.exist(err);
-          res.should.have.status(200);
-          res.should.have.header('content-type', 'application/json');
-
-          res.body.should.be.a('object');
-          res.body.should.have.property('psk_id');
-          res.body.should.have.property('uuid');
-          res.body['psk_id'].should.be.eql(test_psk_id);
-          res.body['uuid'].should.be.eql(test_uuid);
-
-          done();
-        });
+      service.get('/devices/' + test_uuid).then((dataAndResponse) => {
```

Comment:
Instead of replacing old tests entirely, old ones should be modified according to changes and new tests created for new functionality.